tools/toolchains/iar.py

```diff
@@ -126,16 +126,20 @@ def cc_extra(self, object):
         base, _ = splitext(object)
         return ["-l", base + '.s.txt']
 
+    def get_config_option(self, config_header):
+        return ['--preinclude=' + config_header]
+
     def get_compile_options(self, defines, includes, for_asm=False):
-        opts = ['-D%s' % d for d in defines] + ['-f', self.get_inc_file(includes)]
+        opts = ['-f', self.get_inc_file(includes)]
```

Comment:
Uh, whoops. I did not intend to delete that. OTOH, they still seem to propagate to the exporters. I suspect this would break the direct build on IAR, but not an exported project.